editor/frontend/src/TransformElementsTool.ts

```diff
@@ -3,12 +3,160 @@
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
 
-import { Angle, Geometry, Matrix3d, Point3d, Range3d, Transform, Vector3d, YawPitchRollAngles } from "@bentley/geometry-core";
-import { AccuDrawHintBuilder, AngleDescription, BeButtonEvent, CoreTools, DynamicsContext, ElementSetTool, GraphicType, IModelApp, NotifyMessageDetails, OutputMessagePriority, ToolAssistanceInstruction } from "@bentley/imodeljs-frontend";
-import { ColorDef, Frustum, IModelStatus, LinePixels, Placement2d, Placement2dProps, Placement3d } from "@bentley/imodeljs-common";
+import { Angle, Geometry, Matrix3d, Point3d, Transform, Vector3d, YawPitchRollAngles } from "@bentley/geometry-core";
+import { AccuDrawHintBuilder, AngleDescription, BeButtonEvent, CoreTools, DynamicsContext, ElementSetTool, GraphicBranch, GraphicType, IModelApp, IModelConnection, IpcApp, NotifyMessageDetails, OutputMessagePriority, readElementGraphics, RenderGraphic, RenderGraphicOwner, ToolAssistanceInstruction } from "@bentley/imodeljs-frontend";
+import { ColorDef, IModelStatus, LinePixels, PersistentGraphicsRequestProps, Placement2d, Placement2dProps, Placement3d } from "@bentley/imodeljs-common";
 import { DialogItem, DialogItemValue, DialogPropertySyncItem, PropertyDescription } from "@bentley/ui-abstract";
 import { BasicManipulationCommandIpc, editorBuiltInCmdIds } from "@bentley/imodeljs-editor-common";
 import { EditTools } from "./EditTool";
+import { Id64, Id64Arg, Id64String } from "@bentley/bentleyjs-core";
+
+/** @alpha */
+export interface TransformGraphicsData {
+  id: Id64String;
+  placement: Placement2d | Placement3d;
+  graphic: RenderGraphicOwner;
+}
+
+/** @alpha */
+export class TransformGraphicsProvider {
+  public readonly iModel: IModelConnection;
+  public readonly data: TransformGraphicsData[];
+  public readonly pending: Map<Id64String, string>;
+  public readonly prefix: string;
+  /** Chord tolerance to use to stroke the element's geometry in meters. */
+  public chordTolerance = 0.01;
+
+  constructor(iModel: IModelConnection, prefix: string) {
+    this.iModel = iModel;
+    this.prefix = prefix;
+    this.data = new Array<TransformGraphicsData>();
+    this.pending = new Map<Id64String, string>();
+  }
+
+  private getRequestId(id: Id64String): string { return `${this.prefix}-${id}`; }
+  private getToleranceLog10(): number { return Math.log10(this.chordTolerance); }

```

Comment:
Native code will currently reject your tolerance if it is not an integer.

Reply:
Added floor.